pkg/client/client_test.go

```diff
@@ -0,0 +1,236 @@
+package client_test
```

Comment:
I would also argue that anything that runs on a mock net should live in an itest directory outside the hierarchy. The solution for a test that lives in 'client' is to make it more injectable so we can unit test it.

I'm not religious about this though.